global.d.ts

```diff
@@ -0,0 +1,8 @@
+import { LoginType } from "@/lib/schema/schema";
+import "little-state-machine";
+
+declare module "little-state-machine" {
```

Comment:
Can you add some info the description of this pr explaining what `little-state-machine` is and why it is helpful? I'm not very familiar with this library.